quickwit/quickwit-common/src/lib.rs

```diff
@@ -45,6 +45,8 @@ pub mod tower;
 pub mod type_map;
 pub mod uri;
 
+mod semaphore_with_waiter;
```

Comment:
Personally, I'd leave this custom semaphore in `quickwit-ingest` until we need it somewhere else.

Reply:
yes i agree. moving it.